src/main/java/javaslang/control/Validation.java

```diff
@@ -0,0 +1,253 @@
+ *
+ * Validation.Failure:
+ * Validation&lt;List&lt;String&gt;,Integer&gt; valE = Validation.failure(List.of("error1","error2"));
+ * </code>
+ * </pre>
+ *
+ * @param <E> Value type in the case of failure.
+ * @param <T> Value type in the case of success.
+ * @author Eric Nelson
+ * @since 2.0.1
+ * @see <a href="http://eed3si9n.com/learning-scalaz/Validation.html">Validation</a>
+ */
+public interface Validation<E,T> extends Functor<T> {
+
+    static <E,T> Validation<E,T> success(T value) {
+        return new Success<>(value);
+    }
+
+    static <E,T> Validation<E,T> success(Supplier<T> supplier) {
+        return new Success<>(supplier.get());
+    }
+
+    static <E,T> Validation<E,T> failure(E error) {
+        return new Failure<>(error);
+    }
+
+    static <E,T> Validation<E,T> failure(Supplier<E> supplier) {
+        return new Failure<>(supplier.get());
+    }
+
+    boolean isSuccess();
+
+    boolean isFailure();
+
+    Object get();
+
+    void foreach(Consumer<T> f);
+
+    @Override
+    boolean equals(Object o);
+
+    @Override
+    int hashCode();
+
+    @Override
+    String toString();
+
+    default <U> U fold(Function<? super E,? extends U> fail, Function<? super T,? extends U> success) {
+        if(isFailure()) {
+            E v = ((Failure<E,T>) this).get();
+            return fail.apply(v);
+        } else {
+            T v = ((Success<E,T>) this).get();
+            return success.apply(v);
+        }
+    }
+
+    default Validation<T,E> swap() {
+        if(isFailure()) {
+            E v = ((Failure<E,T>) this).get();
+            return Validation.success(v);
+        } else {
+            T v = ((Success<E,T>) this).get();
+            return Validation.failure(v);
+        }
+    }
+
+    @Override
+    default <U> Validation<E,U> map(Function<? super T,? extends U> mapper) {
+        if(isFailure()) {
+            E v = ((Failure<E,T>) this).get();
+            return Validation.failure(v);
+        } else {
+            T v = ((Success<E,T>) this).get();
+            return Validation.success(mapper.apply(v));
+        }
+    }
+
+    default <U,R> Validation<U,R> bimap(Function<? super E,? extends U> fail, Function<? super T,? extends R> success) {
+        if(isFailure()) {
+            E v = ((Failure<E,T>) this).get();
+            return Validation.failure(fail.apply(v));
+        } else {
+            T v = ((Success<E,T>) this).get();
+            return Validation.success(success.apply(v));
+        }
+    }
+
+    default <U> Validation<U,T> leftMap(Function<? super E,? extends U> f) {
+        if(isFailure()) {
+            E v = ((Failure<E,T>) this).get();
+            return Validation.failure(f.apply(v));
+        } else {
+            T v = ((Success<E,T>) this).get();
+            return Validation.success(v);
+        }
+    }
+
+    default <U> Validation<List<E>,U> ap(Validation<List<E>,Function1<T,U>> vv) {
```

Comment:
We need to override Applicative.apply and relax the signature. Impl needs to be figured out then.